src/internal/user_api/mod.rs

```diff
@@ -19,7 +19,7 @@ mod requests;
 #[derive(PartialEq, Debug, Serialize, Deserialize, Clone, Eq, Hash)]
 pub struct UserId(pub(crate) String);
 impl UserId {
-    pub fn id(&self) -> &String {
+    pub fn id(&self) -> &str {
```

Comment:
Is this considered a breaking change? This struct gets exposed externally, right?

Reply:
Yes it is part of our public API and is technically a breaking change. For most (all?) cases `&str` is more idiomatic than `&String` (similar to `&Option<T>` vs `Option<&T>`). If it's actually a breaking change depends on on the usage. Rust provides From and Deref impls that mitigate this. There was one change required to ironoxide-java (`.clone()` -> `.to_string()`)

We undoubtedly have more of these `&String` instances, and I could certainly see the sense in changing them all at once vs piecemeal.